src/js/Controllers/SDLController.js

```diff
@@ -53,25 +56,52 @@ class SDLController {
                 } 
                 return;
             case "GetPolicyConfigurationData":
+                console.log('PTU: Received SDL.GetPolicyConfigData', rpc)
+
                 var urls = JSON.parse(rpc.result.value[0])["0x07"]["default"];
                 var parsed_urls = [];
                 for (const url of urls) {
                     parsed_urls.push({'url': url});
                 }
                 store.dispatch(setURLS(parsed_urls))
                 var state = store.getState()
-                if(flags.ExternalPolicies) {
-                    externalPolicies.pack({            
-                        type: 'PROPRIETARY',
-                        policyUpdateFile: state.system.policyFile,
-                        urls: state.system.urls,
-                        retry: state.system.policyRetry,
-                        timeout: state.system.policyTimeout
-                    })
-                } else {
-                    bcController.onSystemRequest(state.system.policyFile, state.system.urls)
+                
+                let regular_ptu_flow = () => {
+                    if(flags.ExternalPolicies) {
+                        externalPolicies.pack({            
+                            type: 'PROPRIETARY',
+                            policyUpdateFile: state.system.policyFile,
+                            urls: state.system.urls,
+                            retry: state.system.policyRetry,
+                            timeout: state.system.policyTimeout
+                        })
+                    } else {
+                        bcController.onSystemRequest(state.system.policyFile, state.system.urls)
+                    }
+                };
+                
+                if(state.system.ptuWithModemEnabled){
+                    console.log('PTU: Starting PTU over vehicle modem');
+                    let switch_to_regular_ptu_flow = () => {
+                        console.log('PTU: PTU over vehicle modem failed. Switching to PTU over mobile')
+                        store.dispatch(setPTUWithModem(false))
+                        regular_ptu_flow()
+                    };
+
+                    vehicleModem.connectPTUManager(flags.PTUWithModemBackendUrl).then(()=> {
```

Comment:
I feel those are two distinct steps. I chose to connect and requestPTU in the same function but it would still work if we were to open the connection earlier(for example in entry.js) and then simply call the requestPTU function here. Separating out the two steps should give us greater flexibility